metashare/api/models.py

```diff
@@ -338,6 +348,12 @@ def finalize_create_pr(self, error=None):
         else:
             self.notify_error(error)
 
+    def finalize_provision(self):
+        if self.status != TASK_STATUSES["In progress"]:
+            self.status = TASK_STATUSES["In progress"]
```

Comment:
I think we'd want to do this only if a task is currently in `Planned` state. The only time it moves back from `Completed` to `In progress` is when changes are captured, not on provision.

Reply:
We should probably also have a `finalize_commit_changes` that can have this logic, where it always moves to `In progress`.

Reply:
I think that these comments answer each other, and the core confusion is that `finalize_provision` is merely poorly named. What it does is really "set in progress if changes (provision, new changes captured) happen".

Reply:
Right, but I'm saying in addition that the "provision" event should not move a task back from `Completed` to `In progress`, whereas a "changes captured" event should do that.

Reply:
First, under what circumstances could provision happen on a completed task? Second, accepting that it could, I think that _should_ mark it as `In progress`, as it's indicative of a change of state that means it's no longer `Completed`, as far as I can tell.